### DISCUSSION TITLE
Demo table with OpenAI to generate derivative column based on user req in prompt

### DISCUSSION THREAD

**Discussion by MikePapinski** [AntlerVC/firetable#1166] [Idea / Feature Request]
**Is your feature request related to a problem? Please describe.**
As a .NET developer my skills in java script are weak, I would love to have a table that use the  openAI model of chatgpt/ gpt4 to help me create derivate columns, tasks, web hooks, action buttons.

**Describe the solution you'd like**
A simple demo table like for generating stable diffusion images but for code.
1. User select action button, derivative, function, web hook or extension task
2. User define in steps what function should do
3. Custom prompt is being generated and send to the openAI model
4. result code is being checked for compilation/run/errors
5. User makes review/changes to the code
6. User clicks on action button to test if it does what it should do 
7. User copy/paste the code to the proper column type

**Describe alternatives you've considered**
maybe different workflow but the goal is to have quickly generated code based on business objectives in plain text

**Additional context**
of course user provide its own openAI API key

**Comment by BeeBombshell** [comment]
Hey @MikePapinski, this is a truly brilliant idea. We are currently working on developing a demo app that uses ChatGPT in the manner you described. Stay tuned! 🚀
